lib/auth/dbobjectimportrule/dbobjectimportrulev1/service.go

```diff
@@ -0,0 +1,222 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package dbobjectimportrulev1
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobjectimportrule/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/databaseobjectimportrule"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// Backend interface for manipulating DatabaseObjectImportRule resources.
+type Backend interface {
+	services.DatabaseObjectImportRules
+}
+
+// DatabaseObjectImportRuleServiceConfig holds configuration options for
+// the DatabaseObjectImportRules gRPC service.
+type DatabaseObjectImportRuleServiceConfig struct {
+	Authorizer authz.Authorizer
+	Backend    Backend
+	Logger     logrus.FieldLogger
+}
+
+// NewDatabaseObjectImportRuleService returns a new instance of the DatabaseObjectImportRuleService.
+func NewDatabaseObjectImportRuleService(cfg DatabaseObjectImportRuleServiceConfig) (*DatabaseObjectImportRuleService, error) {
+	if cfg.Authorizer == nil {
+		return nil, trace.BadParameter("authorizer is required")
+	}
+	if cfg.Backend == nil {
+		return nil, trace.BadParameter("backend service is required")
+	}
+	if cfg.Logger == nil {
+		cfg.Logger = logrus.WithField(trace.Component, "DatabaseObjectImportRule.service")
```

Comment:
IIRC, the component gets cut off after several characters?

Reply:
Some examples from my journalctl log: `[DB:SERVIC]`, `[UPLOAD]`. `[PROXY:AGE]`. I don't think it's a big deal.

Reply:
I've trimmed the name somewhat.